src/lib/storage/dictionary_segment.hpp

```diff
@@ -25,51 +28,104 @@ class DictionarySegment : public BaseSegment {
   /**
    * Creates a Dictionary segment from a given value segment.
    */
-  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment);
-
-  // SEMINAR INFORMATION: Since most of these methods depend on the template parameter, you will have to implement
-  // the DictionarySegment in this file. Replace the method signatures with actual implementations.
-
-  // return the value at a certain position. If you want to write efficient operators, back off!
-  AllTypeVariant operator[](const ChunkOffset chunk_offset) const override;
-
-  // return the value at a certain position.
-  T get(const size_t chunk_offset) const;
+  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment) {
+    const auto segment_size = base_segment->size();
+    _dictionary = std::make_shared<std::vector<T>>(segment_size);
+
+    for (auto cell_id = 0u; cell_id < segment_size; ++cell_id) {
+      (*_dictionary)[cell_id] = type_cast<T>((*base_segment)[cell_id]);
+    }
+
+    std::sort(_dictionary->begin(), _dictionary->end());
+    const auto first_duplicate_element = std::unique(_dictionary->begin(), _dictionary->end());
```

Comment:
You can give a better name to first_duplicate_element, so that we know that is is a iterator.